ledger/txtail.go

```diff
@@ -80,7 +80,7 @@ type txTail struct {
 	// lastValid, recent, lowWaterMark, roundTailHashes, roundTailSerializedDeltas and blockHeaderData.
 	tailMu deadlock.RWMutex
 
-	lastValid map[basics.Round]map[transactions.Txid]struct{} // map tx.LastValid -> tx confirmed set
+	lastValid map[basics.Round]map[transactions.Txid]basics.Round // map tx.LastValid -> tx confirmed set
```

Comment:
Now that this has a Round in the type twice, it would be good to have a detailed comment.  I believe it is

// lastValid allows looking up all of the transactions that expire in a given round. The map for an expiration round gives the round the transaction was originally confirmed, so it can be found for the `/pending` endpoint.

Here's a tiny optimization that _probably_ isn't worth it.  The newly added `basics.Round` could actually be a uint16 and just encode a delta from the lastValid round back to the confirmation round.

Reply:
I like the deltas suggestion, it would decrease memory foot print from 208 MB (26k txns per bull block * 1000 rounds * 8 bytes) to 52 (26M * 2 bytes)